File: Array/leetcode/medium/TwoSum.py
#https://leetcode.com/problems/two-sum/
def two_sum(nums, target):
    hash_map = {}
    for i in range(len(nums)):
        complement = target - nums[i]
        if complement in hash_map:
            return [hash_map[complement], i]
        hash_map[nums[i]] = i
    return []

test_cases = [
    ([2, 7, 11, 15], 9),
    ([3, 2, 4], 6),
    ([3, 3], 6),
]

for i, (nums, target) in enumerate(test_cases, 1):
    print(f"Test Case {i}:", two_sum(nums, target))

File: Array/leetcode/medium/TwoSum.go
// https://leetcode.com/problems/two-sum/
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    hashMap := make(map[int]int)
    result := []int{-1, -1}
	for i, num := range nums {
        if value, ok := hashMap[target-nums[i]]; ok {
            result[0] = value
            result[1] = i

            return result
        }

        hashMap[num] = i
    }
    return result

}

func twoSumMain() {
    testCases := []struct {
        nums   []int
        target int
    }{
        {[]int{2, 7, 11, 15}, 9},
        {[]int{3, 2, 4}, 6},
        {[]int{3, 3}, 6},
    }

    for i, testCase := range testCases {
        fmt.Printf("Test Case %d: %v\n", i+1, twoSum(testCase.nums, testCase.target))
    }
}

